### PR TITLE
RT2.7 isis_interface_passive_test.go

### PR DESCRIPTION
**Pull request by divyarema**
Initial commit for RT2.7
It resolves [#1615](https://github.com/openconfig/featureprofiles/issues/1615)

### CODE REVIEW

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+## Telemetry Parameter coverage
+
+*   For prefix:
+
+    *   /network-instances/network-instance/protocols/protocol/isis/
+
+*   Parameters:
+
+	*   interfaces/interface/state/passive
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/README.md:67]: This is indented too much

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+}
+
+const (
+	plenIPv4       = 30
+	plenIPv6       = 126
+	ISISInstance   = "DEFAULT"
+	dutAreaAddress = "49.0001"
+	ateAreaAddress = "49.0002"
+	dutSysID       = "1920.0000.2001"
+	v4_metric      = 100
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:46]: do not use underscores in Go names, only camelCase

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+	atePort1attr = attrs.Attributes{
+		Name:    "ATE to DUT port1 ",
+		IPv4:    "192.0.2.2",
+		IPv6:    "2001:db8::192:0:2:2",
+		IPv4Len: plenIPv4,
+		IPv6Len: plenIPv6,
+	}
+)
+
+// configureDUT configures all the interfaces on the DUT.
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:68]: FYI: comments on unexported functions are optional and a lot of these in this test appear to be self-explanatory from the function name

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+func configureDUT(t *testing.T) {
+
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:70]: delete blank line and call t.Helper()

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 5]
```diff
@@ -0,0 +1,399 @@
...
+	gnmi.Replace(t, dut, dc.Interface(i1.GetName()).Config(), i1)
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, dut.Port(t, "port1"))
+	}
+}
+
+// configureIsisDut configures isis configs on DUT.
+func configureIsisDut(t *testing.T, dut *ondatra.DUTDevice, intfName string, dutAreaAddress, dutSysID string) {
+
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:85]: delete blank line and call t.Helper()

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 6]
```diff
@@ -0,0 +1,399 @@
...
+	intf.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV4, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+	intf.GetOrCreateAf(oc.IsisTypes_AFI_TYPE_IPV6, oc.IsisTypes_SAFI_TYPE_UNICAST).Enabled = ygot.Bool(true)
+
+	t.Log("Pushing isis config on DUT")
+	gnmi.Replace(t, dut, configPath.Config(), prot)
+}
+
+// configureATE configures the interfaces and isis protocol on ATE.
+func configureATE(t *testing.T, ate *ondatra.ATEDevice) *ondatra.ATETopology {
+
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:155]: delete blank line and call t.Helper()

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+	t.Log("Pushing config to ATE and starting protocols...")
+	topo.Push(t)
+	topo.StartProtocols(t)
+	return topo
+}
+
+// TestIsisInterfacePassive verifies passive isis interface.
+func TestIsisInterfacePassive(t *testing.T) {
+
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:181]: delete blank line

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 8]
```diff
@@ -0,0 +1,399 @@
...
+		t.Run("Verifying adjacency", func(t *testing.T) {
+			adjacencyPath := statePath.Interface(intfName).Level(2).AdjacencyAny().AdjacencyState().State()
+
+			_, ok := gnmi.WatchAll(t, dut, adjacencyPath, time.Minute, func(val *ygnmi.Value[oc.E_Isis_IsisInterfaceAdjState]) bool {
+				state, present := val.Val()
+				return present && state == oc.Isis_IsisInterfaceAdjState_UP
+			}).Await(t)
+			if !ok {
+				t.Logf("Interface %v has no level 2 isis adjacency", intfName)
+				t.Fatal("No isis adjacency reported.")
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:215]: no need to both log and fail. If there is important information in the log, put it in the failure message

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 9]
```diff
@@ -0,0 +1,399 @@
...
+			mac, err := net.ParseMAC(snpa_address)
+			if !(mac != nil && err == nil) {
+				t.Errorf("FAIL- Expected value for snpa address not found,got %s", snpa_address)
+			}
+			// Checking isis adjacency address families.
+			if got := gnmi.Get(t, dut, adjPath.Nlpid().State()); !cmp.Equal(got, []oc.E_Adjacency_Nlpid{oc.Adjacency_Nlpid_IPV4, oc.Adjacency_Nlpid_IPV6}) {
+				t.Errorf("FAIL- Expected address families not found,got %s,want %s", got, []oc.E_Adjacency_Nlpid{oc.Adjacency_Nlpid_IPV4, oc.Adjacency_Nlpid_IPV6})
+			}
+			// Checking neighbor ipv6 address.
+			ipv6_address := gnmi.Get(t, dut, adjPath.NeighborIpv6Address().State())
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:295]: remove all the underscores from these names and use camelCase instead

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 10]
```diff
@@ -0,0 +1,399 @@
...
+			ip := net.ParseIP(ipv6_address)
+			if !(ip != nil && ip.To16() != nil) {
+				t.Errorf("FAIL- Expected ipv6 address not found,got %s", ipv6_address)
+			}
+			// Checking isis priority.
+			_, present := gnmi.Lookup(t, dut, adjPath.Priority().State()).Val()
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:301]: in Go, it is idiomatic to name these variables "ok" and to put the assignment in the if condition, i.e.

```
if _, ok := gnmi.Lookup(t, dut, adjPath.Priority().State()).Val(); !ok {
	t.Errorf("FAIL- Priority is not present")
}
```
and so on with the rest ...

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 11]
```diff
@@ -0,0 +1,399 @@
...
+			// Checking isis restart suppress.
+			_, checks_present := gnmi.Lookup(t, dut, adjPath.RestartStatus().State()).Val()
+			if !checks_present {
+				t.Errorf("FAIL- Restart suppress not present")
+			}
+		})
+		t.Run("System level counter checks", func(t *testing.T) {
+			// Checking authFail counters.
+			if got := gnmi.Get(t, dut, statePath.Level(2).SystemLevelCounters().AuthFails().State()); got != 0 {
+				t.Errorf("FAIL- Not expecting any authentication key failure,got %d,want %d", got, 0)
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:324]: add a space after the commas in all these error messages

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/README.md (hunk 1 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/README.md]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go (hunk 2 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go (hunk 4 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go (hunk 5 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go (hunk 7 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go (hunk 9 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go (hunk 11 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go (hunk 10 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go (hunk 8 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go]: taken care in latest commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go (hunk 3 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:68]: Noted.

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/README.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
+## RT-2.7: IS-IS Passive is enabled at interface level
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/README.md:1]: this line should only start with a single #. The ## is causing the addrundata check to fail

**Review by greg-dennis — CHANGES_REQUESTED** [review]

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/README.md (hunk 12 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/README.md]: Addressed in latest commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go (hunk 6 above)
Comment (divyarema) [reply feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go]: addressed in recent commit

File: feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go [hunk 13]
```diff
@@ -0,0 +1,393 @@
...
+	t.Log("Pushing interface config on DUT")
+	gnmi.Replace(t, dut, dc.Interface(i1.GetName()).Config(), i1)
+
+	if deviations.ExplicitPortSpeed(dut) {
+		fptest.SetPortSpeed(t, dut.Port(t, "port1"))
+	}
+}
+
+// configureIsisDut configures isis configs on DUT.
+func configureIsisDut(t *testing.T, dut *ondatra.DUTDevice, intfName string, dutAreaAddress, dutSysID string) {
```
Comment (greg-dennis) [inline feature/experimental/isis/ate_tests/isis_interface_passive_test/isis_interface_passive_test.go:84]: go style says "Isis" and "Dut" must both be "ISIS" and "DUT". So technically this must be named configureISISDUT, but maybe just shorten to configureISIS here.

**Review by greg-dennis — CHANGES_REQUESTED** [review]
One style change I missed

**Review by greg-dennis — DISMISSED** [review]
If all the presubmit checks pass, clear to merge

**Review by greg-dennis — APPROVED** [review]

**Review by arulkumarsekar — DISMISSED** [review]

**Review by sachendras — APPROVED** [review]

**Review by arulkumarsekar — APPROVED** [review]

**Pull request merged**